File: leet_easy_c++/leet_067/leet_067b.cpp
class Solution {
public:
    string addBinary(string a, string b) {
        int len1 = a.size(), len2=b.size();
        int cnt=0, carry = 0, tmp = 0;
        string num_str = "";
        int a10, b10;
        while(cnt<len1 || cnt<len2){
            a10 = (a[len1-cnt-1] == '1' && cnt < len1 ? 1 : 0);
            b10 = (b[len2-cnt-1] == '1' && cnt < len2 ? 1 : 0);
            tmp = a10 + b10 + carry;
            if(tmp == 0){ 
                num_str = "0" + num_str; 
                carry = 0;
                cnt++;
            }else if(tmp == 1){
                num_str = "1" + num_str; 
                carry = 0;
                cnt++;
            }else if(tmp == 2){
                num_str = "0" + num_str; 
                carry = 1;
                cnt++;
            }else{  // tmp == 3
                num_str = "1" + num_str; 
                carry = 1;
                cnt++;
            }
        }
        if(carry == 1) num_str = "1" + num_str; 
        return num_str;
    }
};

File: leet_easy_python/leet_067/leet_067b.py
class Solution(object):
    def addBinary(self, a, b):
        carry = 0
        res = ""
        az = a[::-1]
        bz = b[::-1]
        i = 0
        sz = max(len(az), len(bz))
        for i in range(len(az), sz): az +="0" 
        for i in range(len(bz), sz): bz +="0" 
        for aa,bb in zip(az, bz):
            tmp = int(aa) + int(bb) + carry
            carry = 1 if tmp > 1 else 0
            res += str(tmp%2)
        if carry >0: res += "1"      
        return res[::-1]
        
        """
        :type a: str
        :type b: str
        :rtype: str
        """